docs/devsetup/java/maven.md

```diff
@@ -0,0 +1,229 @@
+The Maven page lists [system requirements](https://maven.apache.org/download.cgi),
+which includes a minimum of JDK 8.
+
+## IDE Setup
+
+Users of IDEs should familiarize themselves with the information about how to use Maven at
+the command line,
+which also includes information about Maven builds.
+
+> :point_right: **Note**: In most IDEs, whenever the Maven configurations change (`pom.xml` files),
+you will need to refresh your IDE project using the IDE's Maven plugin/functionality.
+
+Most IDEs represent user settings for different codebases via an IDE-specific notion
+usually called "project" or "workspace".
+The IDE's project/workspace is a separate construct from the Maven configurations for the codebase.
+Therefore, creating a new project for an existing codebase is done by "importing" the codebase,
+in which the IDE creates the project/workspace settings files.
+If you pull updates to the upstream ICU codebase codebase that result in changes to Maven settings,
+then the IDE behavior may be behind until you update your IDE project accordingly.
+
+
+### IntelliJ
+
+IntelliJ does a good job of understanding multi-module Maven projects, 
+including the non-standard configuration here.
+It also recognizes the customized locations of source code files and test code files in the configuration here.
+
+To import into IntelliJ:
+
+1. In IntelliJ, open a new project. 
+  a. Recent versions of IntelliJ provide a dialog box on startup to select a project. Click the "Open" button.)
+2. Select the root `pom.xml` in ICU4J (ex: `<ICU>/icu4j/pom.xml`)
+3. That's it. Note: IntelliJ will take a few minutes to do a one-time indexing of the new source code.
+
+Navigating the source code files between main code and test code, and running tests individually or for an entire module,
+work as they do normally in IntelliJ.
+
+### Eclipse
+
+[Eclipse's Maven plugin](https://eclipse.dev/m2e/)
+works reasonably well and can support the import of a multi-module Maven project.
+These instructions have not yet verified Eclipse's handling of the import of the ICU4J using a Maven build.
+
+### VS Code
+
+VS Code's support of Maven projects is not as robust as IntelliJ's when it comes to the non-standard file layout for sources and tests.
+The Maven support comes from the standard Java extension (which depends on the standard Maven extension) from the extension marketplace.
+Source and test code files are not recognized properly, and it is not clear how to execute the tests.
+
+However, a workaround exists for those who want to use VS Code as their preferred editor and still execute commands to recompile or run tests.
+The workaround relies on invoking the Maven commands in a shell, and using a VS Code extension to create shortcuts within the IDE to invoke those commands.
+
+The extension is [Command Runner](https://marketplace.visualstudio.com/items?itemName=edonet.vscode-command-runner).
```

Comment:
Nit: there is also a really easy way to install from command line:
`--install-extension <ext-id | path>`

There are also nice ways to list, uninstall, etc
(`code --help | grep -i extension`)

Reply:
Ack.